packages/Swiper/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { memo, useCallback, useMemo, useState } from 'react'
+import { func, node, string } from 'prop-types'
+
+import * as S from './styles'
+
+export const Swiper = memo(function Swiper({
+  children,
+  keyPrefix = 'swiper',
+  renderNextButton,
+  renderPaginationItem,
+  renderPrevButton,
+  ...props
+}) {
+  const [pageIdx, setPageIdx] = useState(0)
+
+  const maxSlides = useMemo(() => children.length, [children])
+
+  const nextPage = useCallback(() => {
+    const nextPageIdx = pageIdx + 1
+
+    if (nextPageIdx < maxSlides) {
+      setPageIdx(nextPageIdx)
+    }
+  }, [pageIdx, maxSlides])
+
+  const prevPage = useCallback(() => {
+    const prevPageIdx = pageIdx - 1
+
+    if (prevPageIdx >= 0) {
+      setPageIdx(prevPageIdx)
+    }
+  }, [pageIdx])
```

Comment:
Je pense que les `useMemo` & `useCallback` ne sont pas utile (ils peuvent même rendre le render plus lent 😱).
Kent C. Dodds a un super article à ce sujet: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Merci pour l'article, il éclaire bien les potentiels soucis.
Du coup je suis d'accord que dans ce cas là le useMemo n'est vraiment pas utile (vu qu'il n'y a rien de lourd dedans).
Maybe que dans ce cas précis le useCallback est pas necessaire vu qu'il y a peu de choses en dessous qui pourraient s'update inutilement a chaque render.
Je vais voir pour virer ça.

Par contre, l'article regarde un cas vraiment bien précis sans trop insister sur le contexte qui est très important ! Par exemple dès que tu fais un custom hook ou bien un composant assez haut dans le tree ou bien un composant soumis a de nombreuses updates (des forms par exemple), je pense qu'il faut les utiliser de manière systématique parce que les centaines de rerender que tu causes ont un cout bien bien plus important (bien expérimenté sur WH :D)